### PR TITLE
feat: timer added to the Quiz in LMS

### PR DESCRIPTION
**Pull request by nehasacher143**
**Task Link:** https://app.asana.com/0/1194051119161697/1200048723927509/f

**Screenshots:
Timer Running:**
screenshot-localhost_8005-2021 04 08-12_09_54

**Timer Completed:**
screenshot-localhost_8005-2021 04 08-12_12_18

**Cannot take Quiz:**
screenshot-localhost_8005-2021 04 08-12_12_38

### CODE REVIEW

File: erpnext/education/custom/quiz.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   "print_hide": 0,
+   "print_hide_if_no_value": 0,
+   "print_width": null,
+   "read_only": 0,
+   "report_hide": 0,
+   "reqd": 0,
+   "search_index": 0,
+   "translatable": 0,
+   "unique": 0,
+   "width": null
```
Comment (hrwX) [inline erpnext/education/custom/quiz.json]: Add this in doctype and not custom field

File: erpnext/education/utils.py [hunk 2]
```diff
@@ -232,7 +232,7 @@ def get_quiz(quiz_name, course):
 	student = get_current_student()
 	course_enrollment = get_enrollment("course", course, student.name)
 	status, score, result = check_quiz_completion(quiz, course_enrollment)
-	return {'questions': questions, 'activity': {'is_complete': status, 'score': score, 'result': result}}
+	return {'questions': questions, 'activity': {'is_complete': status, 'score': score, 'result': result}, 'quiz_time': quiz.total_time_for_quiz}
```
Comment (hrwX) [inline erpnext/education/utils.py]:
```suggestion
	return {
		'questions': questions, 
		'activity': {
			'is_complete': status, 
			'score': score, 
			'result': result
		},
		 'quiz_time': quiz.total_time_for_quiz
	}
```

File: erpnext/public/js/education/lms/quiz.js [hunk 3]
```diff
@@ -11,14 +11,44 @@ class Quiz {
...
+				$('#timer').html(" ")
+			}
 		});
 	}
 
+	timer(quiz_time) {
+		let me = this;
+		var minutes = parseInt(quiz_time / 60) % 60;
+		var seconds = quiz_time % 60;
+		var result = (minutes < 10 ? "0" + minutes : minutes) + ":" + (seconds < 10 ? "0" + seconds : seconds);
```
Comment (hrwX) [inline erpnext/public/js/education/lms/quiz.js]: use `let` and not `var`

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/public/js/education/lms/quiz.js [hunk 4]
```diff
@@ -11,14 +11,44 @@ class Quiz {
...
+		if (quiz_time > 0) {
+			$('#timer').html(result + '<span class="timer_text">Time Remaining</span>');
+		}
+
+		if (quiz_time == 0 || quiz_time < 0) {
+			$('#timer').html("Time up");
```
Comment (sahil28297) [inline erpnext/public/js/education/lms/quiz.js]:
```suggestion
			$('#timer').html("Time's Up");
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/public/js/education/lms/quiz.js [hunk 5]
```diff
@@ -11,14 +11,44 @@ class Quiz {
...
 			course: this.course
 		}).then(res => {
-			this.make(res.message)
+			this.make(res.message);
+			if (res.message.activity.is_complete != true) {
+				quiz_time = res.message.quiz_time;
+				this.timer(quiz_time);
+			}
+			else {
+				$('#timer').html(" ")
```
Comment (sahil28297) [inline erpnext/public/js/education/lms/quiz.js]: is the `" "` intentional, should it be `""`?

File: erpnext/public/js/education/lms/quiz.js [hunk 6]
```diff
@@ -11,14 +11,44 @@ class Quiz {
...
+		let me = this;
+		let minutes = parseInt(quiz_time / 60) % 60;
+		let seconds = quiz_time % 60;
+		let result = (minutes < 10 ? "0" + minutes : minutes) + ":" + (seconds < 10 ? "0" + seconds : seconds);
+		if (quiz_time > 0) {
+			$('#timer').html(result + '<span class="timer_text">Time Remaining</span>');
+		}
+
+		if (quiz_time == 0 || quiz_time < 0) {
+			$('#timer').html("Time's up");
```
Comment (sahil28297) [inline erpnext/public/js/education/lms/quiz.js]:
```suggestion
			$('#timer').html("Time's Up");
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**